hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1259,16 +1283,18 @@ public void quasiCloseContainer(Container container, String reason)
                 .getContainerID() + " while in " + state + " state.", error);
       }
       container.quasiClose();
-      ContainerLogger.logQuasiClosed(container.getContainerData(), reason);
-      sendICR(container);
     } finally {
       container.writeUnlock();
     }
+    createContainerMerkleTree(container);
+    ContainerLogger.logQuasiClosed(container.getContainerData(), reason);
+    sendICR(container);
   }
 
   @Override
   public void closeContainer(Container container)
       throws IOException {
+    createContainerMerkleTree(container);
```

Comment:
Why does this method not follow the same pattern as the others for when to call tree creation? Usually it won't be necessary since `markContainerForClose` would already be called but it's still nice to have since we already have a file existence check to remove duplicate work.